File: ESTRUTURAS/1260/main.py
from sys import stdin


def get_trees():
    trees = {}
    for tree in map(str.strip, stdin):
        if not tree:
            break
        if not tree in trees:
            trees[tree] = 0
        trees[tree] += 1
    return trees


def formatted_print(c, trees):
    if c:
        print()
    total = sum(trees.values())
    for tree in sorted(trees.keys()):
        print(f'{tree} {100 * trees[tree] / total:.4f}')


n = int(input())
input()
for c in range(n):
    formatted_print(c, get_trees())

File: ESTRUTURAS/1260/main.rb
def get_trees
  trees = {}
  while (tree = gets&.strip)
    break if tree.empty?
    trees[tree] = 0 if !trees.key?(tree)
    trees[tree] += 1
  end
  trees
end

def formatted_print(c, trees)
  puts if c.nonzero?
  total = trees.values.sum
  trees.keys.sort.each do |tree|
    puts "#{tree} #{"%.4f" % (100.0 * trees[tree] / total)}"
  end
end

n = gets.to_i
gets
n.times do |c|
  formatted_print(c, get_trees)
end
